src/httpd_artwork.c

```diff
@@ -0,0 +1,293 @@
+/*
+ * Copyright (C) 2018 Ray/whatdoineed2do <whatdoineed2do@nospam.gmail.com>
+ *
+ * Adapted from httpd_jsonapi.c:
+ * Copyright (C) 2017 Christian Meffert <christian.meffert@googlemail.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+
+#ifdef HAVE_CONFIG_H
+# include <config.h>
+#endif
+
+#include <regex.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "httpd_artwork.h"
+#include "db.h"
+#include "library.h"
+#include "logger.h"
+#include "misc.h"
+#include "player.h"
+#include "artwork.h"
+
+
+static int
+artwork_to_evbuf(struct httpd_request *hreq, struct db_group_info *dbgri, uint32_t *dbmfi_id)
+{
+  int ret = 0;
+  int len = 0;
+  const char* ctype = "invalid";
+  char clen[32];
+  struct evkeyvalq *headers;
+
+  if ( (dbgri == NULL && dbmfi_id == NULL) || (dbgri && dbmfi_id)) 
+    {
+      return -1;
+    }
+
+  if (dbgri)
+    {
+      int  gpid;
+      safe_atoi32(dbgri->id, &gpid);
+      if ( (ret = artwork_get_group(hreq->reply, gpid, 600, 600)) < 0)
```

Comment:
Default dimensions are fine, but there should be params for requesting other sizes